server/services/RollupService.ts

```diff
@@ -322,71 +273,63 @@ export default class RollupService {
   ): Promise<IKibanaResponse<ServerResponse<GetRollupsResponse>>> => {
     try {
       const { from, size, search, sortDirection, sortField } = request.query as {
-        from: string;
-        size: string;
+        from: number;
+        size: number;
         search: string;
         sortDirection: string;
         sortField: string;
       };
 
-      const rollupSorts: RollupsSort = {
-        id: "rollup.rollup_id.keyword",
-        "rollup.rollup.description": "rollup.description.keyword",
-        "rollup.rollup.last_updated_time": "rollup.last_updated_time",
-      };
       const params = {
-        index: INDEX.OPENDISTRO_ISM_CONFIG,
-        seq_no_primary_term: true,
-        body: {
-          size,
-          from,
-          sort: rollupSorts[sortField] ? [{ [rollupSorts[sortField]]: sortDirection }] : [],
-          query: {
-            bool: {
-              filter: [{ exists: { field: "rollup" } }],
-              must: getMustQuery("rollup.rollup_id", search),
-            },
-          },
-        },
+        from,
+        size,
+        search,
+        sortField,
+        sortDirection,
       };
 
       const { callAsCurrentUser: callWithRequest } = this.esDriver.asScoped(request);
-      const searchResponse: SearchResponse<any> = await callWithRequest("search", params);
-      const totalRollups = searchResponse.hits.total.value;
-
-      const rollups = searchResponse.hits.hits.map((hit) => ({
-        _seqNo: hit._seq_no as number,
-        _primaryTerm: hit._primary_term as number,
-        _id: hit._id,
-        rollup: hit._source,
+      const getRollupResponse: GetRollupsResponse = await callWithRequest("ism.getRollups", params);
+      const totalRollups = getRollupResponse.total_rollups;
+      const rollups = getRollupResponse.rollups.map((rollup) => ({
+        _seqNo: rollup._seqNo as number,
+        _primaryTerm: rollup._primaryTerm as number,
+        _id: rollup._id,
+        rollup: rollup.rollup,
         metadata: null,
       }));
+
+      // Call getExplain if any rollup job exists
       if (totalRollups) {
+        // Concat rollup job ids
         const ids = rollups.map((rollup) => rollup._id).join(",");
-        const explainResponse = await this.explainRollup(context, request, response, ids);
-        if (explainResponse.payload.ok) {
+        const explainResponse = await callWithRequest("ism.explainRollup", { rollupId: ids });
+        if (!explainResponse.error) {
           rollups.map((item) => {
-            item.metadata = explainResponse.payload.response[item._id];
+            item.metadata = explainResponse[item._id];
           });
           return response.custom({
             statusCode: 200,
-            body: { ok: true, response: { rollups: rollups, totalRollups: totalRollups, metadata: explainResponse } },
+            body: { ok: true, response: { rollups: rollups, total_rollups: totalRollups, metadata: explainResponse } },
```

Comment:
Any reason for changing this to snake_case?

Reply:
It was a few changes after adding type definition of `GetRollupsResponse` in line 293, the interface definition was changed to snake_case, too. 
Since other functions does not define the type of received response, had just removed the type definition in line 292 making it `const getRollupResponse = await ...` and changed snake_case back to camelCase.